cmd/cloud.go

```diff
@@ -60,56 +60,71 @@ This will execute the test on the k6 cloud service. Use "k6 login cloud" to auth
 		Example: `
         k6 cloud script.js`[1:],
 		Args: exactArgsWithMsg(1, "arg should either be \"-\", if reading script from stdin, or a path to a script file"),
-		RunE: func(cmd *cobra.Command, args []string) error {
-			// we specifically first parse it and return an error if it has bad value and then check if
-			// we are going to set it  ... so we always parse it instead of it breaking the command if
-			// the cli flag is removed
-			if showCloudLogsEnv, ok := os.LookupEnv("K6_SHOW_CLOUD_LOGS"); ok {
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// TODO: refactor (https://github.com/loadimpact/k6/issues/883)
+			//
+			// We deliberately parse the env variables, to validate for wrong
+			// values, even if we don't subsequently use them (if the respective
+			// CLI flag was specified, since it has a higher priority).
+			if showCloudLogsEnv, ok := globalState.envVars["K6_SHOW_CLOUD_LOGS"]; ok {
 				showCloudLogsValue, err := strconv.ParseBool(showCloudLogsEnv)
 				if err != nil {
 					return fmt.Errorf("parsing K6_SHOW_CLOUD_LOGS returned an error: %w", err)
 				}
 				if !cmd.Flags().Changed("show-logs") {
-					globalFlags.showCloudLogs = showCloudLogsValue
+					showCloudLogs = showCloudLogsValue
 				}
 			}
+
+			if exitOnRunningEnv, ok := globalState.envVars["K6_EXIT_ON_RUNNING"]; ok {
```

Comment:
These two blocks look similar, it can be a small helper

Reply:
they are similar, but it's just 2 blocks, and we plan to rewrite this whole config mess with croconf (hopefully soon)... this repetition can be extra incentive